go/chat/server.go

```diff
@@ -1310,6 +1316,28 @@ func (h *Server) PostLocalNonblock(ctx context.Context, arg chat1.PostLocalNonbl
 	}, nil
 }
 
+// If we are superseding an ephemeral message, we have to set the
+// ephemeralMetadata on this superseder message.
+func (h *Server) getSupersederEphemeralMetadata(ctx context.Context, uid gregor1.UID, convID chat1.ConversationID, msg chat1.MessagePlaintext) (metadata *chat1.MsgEphemeralMetadata, err error) {
+	switch msg.ClientHeader.MessageType {
+	case chat1.MessageType_EDIT, chat1.MessageType_ATTACHMENTUPLOADED:
+	default:
+		return msg.ClientHeader.EphemeralMetadata, nil
+	}
+
+	messages, err := h.G().ChatHelper.GetMessages(ctx, uid, convID,
```

Comment:
Use `ConvSource`

Reply:
when should `ChatHelper` be used in general? `GetMessages` just wraps a `ConvSource` call or is there a different method on `ConvSource` i should look at? https://github.com/keybase/client/blob/master/go/chat/helper.go#L218